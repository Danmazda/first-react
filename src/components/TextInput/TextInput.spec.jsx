import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { TextInput } from '.';

describe('<TextInput />', () => {
  it('should render TextInput', () => {
    const fn = jest.fn();
    render(<TextInput handleInput={fn} />);
    const input = screen.getByPlaceholderText(/search/i);
    expect(input).toBeInTheDocument();
  });
  it('should call handleInput on each keypress', () => {
    const fn = jest.fn();
    render(<TextInput handleInput={fn} />);
    const input = screen.getByPlaceholderText(/search/i);
    const value = 'the value';
    userEvent.type(input, value);
    expect(input.value).toBe(value);
    expect(fn).toHaveBeenCalledTimes(value.length);
  });
  it('should match the snapshot', () => {
    const fn = jest.fn();
    const { container } = render(<TextInput handleInput={fn} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
