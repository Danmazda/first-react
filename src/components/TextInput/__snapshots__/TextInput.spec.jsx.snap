// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should match the snapshot 1`] = `
<input
  class="text-input"
  placeholder="Search"
  type="search"
  value=""
/>
`;
